src/UI/Art/SpriteArtist/PlayerSetupCommanderArtist.java

```diff
@@ -305,4 +333,498 @@ public BufferedImage update(CommanderInfo coInfo, Color color)
+  {
+    final CommanderInfo selectedInfo = myControl.cmdrInfos.get(selectedCO);
+
+    ArrayList<Integer>[][] allCmdrLists;
+    ArrayList<COSpriteSpec> outerBinColorSpec = new ArrayList<>();
+    ArrayList<COSpriteSpec> binColorSpec = new ArrayList<>();
+    int outerMax, outerSel;
+    if( sortByGameThenFaction )
+    {
+      outerMax = UIUtils.SourceGames.values().length;
+      outerSel = selectedInfo.game.ordinal();
+
+      // outer bin = game
+      for( SourceGames game : UIUtils.SourceGames.values() )
+        outerBinColorSpec.add(game.uiColorSpec);
+      // Factions for both
+      allCmdrLists = myControl.cosByGameFaction;
+      for( COSpriteSpec spec : myControl.canonFactions )
+        binColorSpec.add(spec);
+    }
+    else
+    {
+      outerMax = myControl.canonFactions.length;
+      outerSel = 0;
+      for( ; outerSel < myControl.canonFactions.length; ++outerSel )
+        if( selectedInfo.baseFaction == myControl.canonFactions[outerSel] )
+          break;
+      // outer bin = faction
+      for( COSpriteSpec spec : myControl.canonFactions )
+        outerBinColorSpec.add(spec);
+      // Games for both
+      allCmdrLists = myControl.cosByFactionGame;
+      for( SourceGames game : UIUtils.SourceGames.values() )
+        binColorSpec.add(game.uiColorSpec);
+    }
+
+    cmdrBins = new ArrayList<>();
+
+    for( int outerIndex = 0; outerIndex < outerMax; ++outerIndex )
+    {
+      COSpriteSpec outerSpec = outerBinColorSpec.get(outerIndex);
+      Color[] outerPalette = UIUtils.defaultMapColors;
+      String outerName = (!sortByGameThenFaction)? " " : "MISC"; // Faction name is handled by the inner title
+      if( Color.LIGHT_GRAY != outerSpec.color )
+      {
+        outerPalette = UIUtils.getMapUnitColors(outerSpec.color).paletteColors;
+        if( sortByGameThenFaction ) // Our outer category is games, so just pull the hardcoded faction name
+          outerName = outerSpec.faction.name;
+      }
+      OuterCategoryPanel outerCategory = new OuterCategoryPanel(outerPalette, outerName);
+
+      for( int innerIndex = 0; innerIndex < allCmdrLists.length; ++innerIndex )
+      {
+        if( 0 >= allCmdrLists[outerIndex][innerIndex].size() )
+          continue;
+
+        ArrayList<Integer> innerBin = new ArrayList<>(allCmdrLists[outerIndex][innerIndex]);
+        final COSpriteSpec spriteSpec = binColorSpec.get(innerIndex);
+        Color[] palette = UIUtils.defaultMapColors;
+        String canonName = "MISC";
+        if( Color.LIGHT_GRAY != spriteSpec.color )
+        {
+          if( sortByGameThenFaction )
+          {
+            palette = UIUtils.getMapUnitColors(spriteSpec.color).paletteColors;
+            canonName = UIUtils.getCanonicalFactionName(spriteSpec);
+          }
+          else // Our inner category is games, so steal stuff from the outer category
+          {
+            palette   = outerPalette;
+            canonName = outerSpec.faction.name + ": " + spriteSpec.faction.name;
```

Comment:
Interesting choice to use the canonical name ("Green Earth") when sorting by game, but just using the faction name ("Earth") when sorting by faction. Any particular reason?

Reply:
Uhhh. Being blind, probably